src/CodegenFactoryTrait.php

```diff
@@ -179,6 +179,12 @@
     return new CodegenShape($this->getConfig(), $attrs);
   }
 
+  final public function codegenShape_FUTURE(
+    vec<CodegenShapeMember> $members = vec([]),
```

Comment:
Maybe:

```
SomeEnum $allow_undefined_fields,
CodegenShapeMember ....$members
```

Reply:
isn't `...$members` implicitly `array<CodegenShapeMember>`? not sure if this breaks the no new usage of `array`